### ISSUE TITLE
Gradle 5.0-milestone-1 throws an exception when accessing properties in settings.gradle

### ISSUE THREAD

**Issue by ericparton** [gradle/gradle#7009]
### Expected Behavior

This statement in the settings.gradle should not cause gradle to immediately throw an exception
```
if (properties.someProjectProperty != 'true') {
    include 'subproject'
}
```
"someProjectProperty" is a project property that is set to true in the root gradle.properties
### Current Behavior

```
FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/ericp/Projects/someproject/settings.gradle' line: 67

* What went wrong:
A problem occurred evaluating settings 'someproject'.
> org.gradle.initialization.DefaultSettings_Decorated.getConvention()Lorg/gradle/api/plugins/Convention;

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s

```
### Steps to Reproduce (for bugs)
Add a conditional statement to settings.gradle that uses a project property

### Your Environment

Gradle 5.0-milestone-1

**Comment by big-guy** [comment]
I can confirm this change in behavior.  I think it's due to https://github.com/gradle/gradle/pull/6685 

@eskatos want to take a look?  I think `DefaultSettings` has to also implement `HasConvention`.

At the same time... it's a little funky that the way this worked was relying on the Groovy metaclass stuff.  It might be worth adding something that's reachable from Kotlin too.

**Comment by eskatos** [comment]
Fixed in #7065
